packages/google_maps_flutter/lib/src/marker_updates.dart

```diff
@@ -0,0 +1,186 @@
+@immutable
+class MarkerUpdates {
+  @visibleForTesting
+  MarkerUpdates.internal({
+    @required this.markerUpdates,
+  }) : assert(markerUpdates != null);
+
+  /// Computes [MarkerUpdates] given previous and current [Marker]s.
+  factory MarkerUpdates.from(Set<Marker> previous, Set<Marker> current) {
+    if (previous == null) {
+      previous = Set<Marker>.identity();
+    }
+
+    if (current == null) {
+      current = Set<Marker>.identity();
+    }
+
+    final Map<MarkerId, Marker> previousMarkers = _toMap(previous);
+    final Map<MarkerId, Marker> currentMarkers = _toMap(current);
+
+    final Set<MarkerId> allMarkerIds =
+        previousMarkers.keys.toSet().union(currentMarkers.keys.toSet());
+
+    final Set<MarkerUpdate> markerUpdates = allMarkerIds.map(
+      (MarkerId markerId) {
+        final bool inCurrent = currentMarkers.containsKey(markerId);
+        final bool inPrevious = previousMarkers.containsKey(markerId);
+
+        if (inCurrent && inPrevious) {
+          return MarkerUpdate._update(
+            oldMarker: previousMarkers[markerId],
+            newMarker: currentMarkers[markerId],
+          );
+        } else if (inCurrent) {
+          return MarkerUpdate._add(currentMarkers[markerId]);
+        } else if (inPrevious) {
+          return MarkerUpdate._remove(markerId);
+        } else {
+          throw FlutterError("Unknown markerId: " + markerId.value);
+        }
+      },
+    ).toSet();
+
+    return MarkerUpdates.internal(markerUpdates: markerUpdates);
+  }
+
+  final Set<MarkerUpdate> markerUpdates;
+
+  dynamic _toJson() {
+    return markerUpdates
+        .map<dynamic>((MarkerUpdate update) => update._toMap())
+        .toList();
+  }
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is MarkerUpdates &&
+          runtimeType == other.runtimeType &&
+          markerUpdates == other.markerUpdates;
+
+  @override
+  int get hashCode => markerUpdates.hashCode;
+
+  @override
+  String toString() {
+    return 'MarkerUpdates{markerUpdates: $markerUpdates}';
+  }
+}
+
+/// [Marker] update event with the changes.
+@immutable
+class MarkerUpdate {
+  @visibleForTesting
+  MarkerUpdate.internal({
+    @required this.updateEventType,
+    @required this.markerId,
+    this.changes,
+    this.newMarker,
+  })  : assert(updateEventType != null),
+        assert(markerId != null),
+        assert(markerId.value != null);
+
+  factory MarkerUpdate._remove(MarkerId markerId) {
+    return MarkerUpdate.internal(
+      updateEventType: MarkerUpdateEventType.remove,
+      markerId: markerId,
+    );
+  }
+
+  factory MarkerUpdate._add(Marker newMarker) {
+    return MarkerUpdate.internal(
+      updateEventType: MarkerUpdateEventType.add,
+      markerId: newMarker.markerId,
+      changes: newMarker,
+      newMarker: newMarker,
+    );
+  }
+
+  /// TODO(iskakaushik): Diff is sufficient, don't need to send the whole update.
```

Comment:
TODOs should include a link to a GitHub issue otherwise they are not in our tracking system,.

Reply:
Will make an issue for this.